core/simd/simd.odin

```diff
@@ -1,182 +1,2273 @@
+Check if values of SIMD vector are greater than or equal the values of another
+SIMD vector.
+
+This procedure checks each pair of lanes from vectors `a` and `b` for whether
+lane of `a` is greater than or equal to the lane of `b`, and if so, the
+corresponding lane of the result vector will have a value with all bits set
+(`0xff..ff`). Otherwise the lane of the result vector will have the value `0`.
+
+Inputs:
+- `a`: An integer or a float vector.
+- `b`: An integer or a float vector.
+
+Returns:
+- A vector of unsigned integers of the same size as the input vector's lanes,
+containing comparison results for each lane.
+
+**Operation**:
+
+	for i in 0 ..< len(res) {
+		if a[i] >= b[i] {
+			res[i] = unsigned(-1)
+		} else {
+			res[i] = 0
+		}
+	}
+	return res
+
+Example:
+
+	   +-------+-------+-------+-------+
+	a: |   0   |   1   |   2   |   3   |
+	   +-------+-------+-------+-------+
+	   +-------+-------+-------+-------+
+	b: |   0   |   2   |   2   |   2   |
+	   +-------+-------+-------+-------+
+	res:
+	   +-------+-------+-------+-------+
+	   | 0xff  | 0x00  | 0xff  | 0xff  |
+	   +-------+-------+-------+-------+
+*/
 lanes_ge :: intrinsics.simd_lanes_ge
 
+/*
+Perform a gather load into a SIMD vector.
+
+A *gather* operation is memory load operation, that loads values from an vector
+of addresses into a single value vector. This can be used to achieve the
+following results:
 
-// Gather and Scatter intrinsics
+- Accessing every N'th element of an array (strided access)
+- Access of elements according to some computed offsets (indexed access).
+- Access of elements in a different order (shuffling access).
+
+When used alongside other SIMD procedures in order to compute the offsets
+for the `ptr` and `mask` parameters.
+
+Inputs:
+- `ptr`: A vector of memory locations. Each pointer points to a single value,
+	of a SIMD vector's lane type, that will be loaded into the vector. Pointer
+	in this vector can be `nil` or any other invalid value, if the corresponding
+	value in the `mask` parameter is zero.
+- `val`: A vector of values that will be used at corresponding positions
+	of the result vector, if the corresponding memory location has been
+	masked out.
+- `mask`: A vector of booleans or unsigned integers that determines which memory
+	locations to read from. If the value at an index has the value true
+	(lowest bit set), the value at that index will be loaded into the result
+	vector from the corresponding memory location in the `ptr` vector. Otherwise
+	the value will be loaded from the `val` vector.
+
+Returns:
+
+This procedure returns a vector with all values from unmasked indices
+loaded from the pointer vector `ptr`, and all values from masked indices loaded
+from the value vector `val`.
+
+**Operation**:
+
+	for i in 0 ..< len(res) {
+		if mask[i]&1 == 1 {
+			res[i] = ptr[i]^
+		} else {
+			res[i] = val[i]
+		}
+	}
+	return res
+
+Example:
+
+Example below loads 2 lanes of values from 2 lanes of float vectors, `v1` and
```

Comment:
Text describing an example should go above `Example:` or be indented and a comment because it is interpreted as a code block. Same comment for all other following docs with examples.